src/cache.jl

```diff
@@ -17,22 +17,26 @@ function alg_cache{T,U}(alg::MIRK4, S::BVPSystem{T,U})
 end
 =#
 
-struct MIRK4GeneralCache{kType} <: GeneralMIRKCache
```

Comment:
```julia
for i in 1:(N - 1)
    h = x[i + 1] - x[i]
    # Update K
    update_K!(S, cache, TU, i, h)
    for r in 1:s
        x_new = x[i] + c[r] * h
        y_new = (1 - v[r]) * y[i] + v[r] * y[i + 1]
        if r > 1
            y_new += h * sum(j -> X[r, j] * K[j], 1:(r - 1))
        end
        temp = zeros(Float64, M)
        fun!(temp, y_new, S.p, x_new)
        K[r] = temp[:]
    end
    # Update residual
    residual[i] = y[i + 1] - y[i] - h * sum(j -> b[j] * K[j], 1:s)
    cache.k_discrete[i, :] = K[:]
end
```

The MIRK cache stores the discrete stages from evaluating ```PHI``` in the nonlinear solving process. Since we only need discrete stages from the final nonlinear solving step, I think the previous steps can be omitted, but I didn't figure out an executable way, e.g. judging if this step is the last time we evaluate ```PHI```🤔.

Reply:
It's mutating the values but not the cache itself